--- a/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
@@ -480,6 +480,35 @@
 				interrupts = <GIC_SPI 218 IRQ_TYPE_EDGE_RISING>;
 				status = "disabled";
 			};
+
+			phy@78000 {
+				compatible = "amlogic,meson-gxl-usb2-phy";
+				#phy-cells = <0x0>;
+				reg = <0x0 0x78000 0x0 0x20>;
+				status = "okay";
+				linux,phandle = <0x2d>;
+				phandle = <0x2d>;
+			};
+
+			phy@78020 {
+				compatible = "amlogic,meson-gxl-usb2-phy";
+				#phy-cells = <0x0>;
+				reg = <0x0 0x78020 0x0 0x20>;
+				status = "okay";
+				linux,phandle = <0x2e>;
+				phandle = <0x2e>;
+			};
+
+			phy@78040 {
+				compatible = "amlogic,meson-gxl-usb2-phy";
+				#phy-cells = <0x0>;
+				reg = <0x0 0x78040 0x0 0x20>;
+				status = "disabled";
+				linux,phandle = <0x2f>;
+				phandle = <0x2f>;
+			};
+
+
 		};
 
 		vpu: vpu@d0100000 {
@@ -529,5 +558,72 @@
 				reg = <1>;
 			};
 		};
-	};
+
+		usb@c9000000 {
+			compatible = "snps,dwc3";
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+			reg = <0x0 0xc9000000 0x0 0x100000>;
+			interrupts = <0x0 0x1e 0x4>;
+			dr_mode = "host";
+			maximum-speed = "high-speed";
+			snps,dis_u2_susphy_quirk;
+			status = "okay";
+
+			roothub@0 {
+				compatible = "usb1d6b,3", "usb1d6b,2";
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+				reg = <0x0>;
+
+				port@1 {
+					reg = <0x1>;
+					phys = <0x2d>;
+					phy-names = "usb2-phy";
+				};
+
+				port@2 {
+					reg = <0x2>;
+					phys = <0x2e>;
+					phy-names = "usb2-phy";
+				};
+
+				port@3 {
+					reg = <0x3>;
+					phys = <0x2f>;
+					phy-names = "usb2-phy";
+				};
+			};
+		};
+
+		sound {
+			compatible = "simple-audio-card";
+			simple-audio-card,name = "meson-gxm-q200";
+
+			simple-audio-card,dai-link@0 {
+
+				plat {
+					sound-dai = <0x30>;
+				};
+
+				cpu {
+					sound-dai = <0x31>;
+				};
+
+				codec {
+					sound-dai = <0x32>;
+				};
+			};
+		};
+
+
+	};
+
+	regulator-usb-pwrs {
+		compatible = "regulator-fixed";
+		regulator-name = "USB_PWR";
+		regulator-min-microvolt = <0x4c4b40>;
+		regulator-max-microvolt = <0x4c4b40>;
+	};
+
 };
